### PR TITLE
Bump HeatpumpIR, add protocols, remove IRremoteESP8266

### PR DESCRIPTION
**Pull request by nagyrobi**
# What does this implement/fix?

- Bump HeatpumpIR to 1.0.27: https://github.com/ToniA/arduino-heatpumpir/commits/1.0.27
- add protocols:
  ```
  airway
  bgh_aud
  panasonic_altdke
  vaillantvai8
  r51m
  ```
- remove IRremoteESP8266 dependency as it's not required anymore (see https://github.com/ToniA/arduino-heatpumpir/pull/175#issuecomment-2190544978 and https://github.com/esphome/esphome/pull/6955#issuecomment-2188338667), allowing usage on platforms like BK72XX.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#3996

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [x] ESP8266
- [ ] RP2040
- [x] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by nagyrobi** [comment]
Tested on an ESP3266, HeatpumpIR 1.0.27 pulls in IRremoteESP8266 on its own:

```
...
Processing ir-1-halo (board: esp01_1m; framework: arduino; platform: platformio/espressif8266@4.2.1)
--------------------------------------------------------------------------------
Library Manager: Installing esphome/noise-c @ 0.1.4
INFO Installing esphome/noise-c @ 0.1.4
Unpacking  [####################################]  100%
Library Manager: noise-c@0.1.4 has been installed!
INFO noise-c@0.1.4 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/libsodium @ 1.10018.1
INFO Installing esphome/libsodium @ 1.10018.1
Unpacking  [####################################]  100%
Library Manager: libsodium@1.10018.1 has been installed!
INFO libsodium@1.10018.1 has been installed!
Library Manager: Installing tonia/HeatpumpIR @ 1.0.27
INFO Installing tonia/HeatpumpIR @ 1.0.27
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: HeatpumpIR@1.0.27 has been installed!
INFO HeatpumpIR@1.0.27 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing crankyoldgit/IRremoteESP8266 @ ~2.8.4
INFO Installing crankyoldgit/IRremoteESP8266 @ ~2.8.4
Unpacking  [####################################]  100%
Library Manager: IRremoteESP8266@2.8.6 has been installed!
INFO IRremoteESP8266@2.8.6 has been installed!
HARDWARE: ESP8266 80MHz, 80KB RAM, 1MB Flash
Dependency Graph
|-- ESP8266WiFi @ 1.0
|-- ESP8266mDNS @ 1.2
|-- noise-c @ 0.1.4
|-- HeatpumpIR @ 1.0.27
Compiling .pioenvs/ir-1-halo/src/esphome/components/api/api_connection.cpp.o
...
```

File: tests/components/heatpumpir/test.esp32-ard.yaml [hunk 1]
```diff
@@ -1,19 +1,12 @@
...
     horizontal_default: left
     vertical_default: up
-    name: HeatpumpIR Climate
-    min_temperature: 18
-    max_temperature: 30
-  - platform: heatpumpir
-    protocol: greeyt
-    horizontal_default: left
-    vertical_default: up
-    name: HeatpumpIR Climate
```
Comment (jesserockz) [inline tests/components/heatpumpir/test.esp32-ard.yaml]: You shouldnt remove these if they still work. Just add new ones.

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: tests/components/heatpumpir/test.esp32-ard.yaml (hunk 1 above)
Comment (nagyrobi) [reply tests/components/heatpumpir/test.esp32-ard.yaml]: I just moved them to the other framework test files,
added back...

**Review by jesserockz — CHANGES_REQUESTED** [review]
You are still removing protocols from the test YAML fiels instead of just adding new ones. Adding entire new test files adds about 5 mins to testing time, adding new climate models to the list adds about 5 seconds =)
Just keep them all there and add more if needed. No need to remove any old protocols unless they are removed from the library itself and no longer compile.

**Comment by nagyrobi** [comment]
I was just trying to optimize a bit, because I thought testing for the exact same component configuration repeatedly would be redundant for each framework; considered a performance gain to distribute some different configuration possibilities over the config files.
As these should actually behave exactly the same at ESPHome level, the difference is only at library level. 

Restored the original files and only added new protocols to the new file. Or would you like to see exactly the same tests for each test file?

**Comment by jesserockz** [comment]
Ideally, we would have a case for as many configuration cases as possible, but that can be a bit overkill. I just don't see the point in removing the ones that work just fine. Just good to add the newly supported ones. Looks good now

**Review by jesserockz — APPROVED** [review]

**Pull request merged**